lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/FunctionalChaosSobolIndices.cxx

```diff
@@ -65,111 +66,118 @@ inline bool varianceComparison(const std::pair<UnsignedInteger, Scalar> &a, cons
-      for (UnsignedInteger k = 0; k < multiIndices.getSize(); ++ k) degreeI += multiIndices[k];
-
-      maxdegree = std::max(maxdegree, degreeI);
-      const Scalar varianceRatio = coefI * coefI / variance[m];
-      varianceOrder.push_back(std::pair<UnsignedInteger, Scalar >(i, varianceRatio));
-    }
-
-    // sort basis terms by descending variance contribution
-    std::sort(varianceOrder.begin(), varianceOrder.end(), varianceComparison);
-
+    if (m > maximumNumberOfOutput) break;
     if (outputDimension > 1)
       oss << "Marginal: " << m << "\n";
 
+    const Point partOfVariance(getPartOfVariance(m));
+    const Sample partOfVarianceSample(Sample::BuildFromPoint(partOfVariance));
+    const Indices order(partOfVarianceSample.argsort(false));
+
     // table of part of variance for each basis term
-    oss << "Index   | Multi-indice                  | Part of variance  \n";
-    oss << String(60, '-') << "\n";
-    for (UnsignedInteger i = 0; i < varianceOrder.size(); ++ i)
+    // Print header
+    oss << "| Index |"
+        << padString(" Multi-index") << "|"
+        << padString(" Part of variance") << "|"
+        << "\n";
+    // Print dashes
+    oss << "|-------|";
+    oss << String(columnWidth, '-') << "|"
+        << String(columnWidth, '-') << "|\n";
+    // Print table content
+    for (UnsignedInteger i = 0; i < basisSize; ++ i)
     {
-      // stop when the variance contribution becomes less than epsilon
-      if (varianceOrder[i].second < ResourceMap::GetAsScalar("FunctionalChaosSobolIndices-VariancePartThreshold"))
-        break;
+      const UnsignedInteger sortedIndex = order[i];
+      const UnsignedInteger rankIndex = indices[sortedIndex];
+      // stop when the part of variance becomes less than epsilon
+      if (partOfVariance[sortedIndex] < varianceThreshold) break;      
+      Indices multiIndices(enumerateFunction(rankIndex));
 
-      Indices multiIndices(enumerateFunction(indices[varianceOrder[i].first]));
+      oss << "|" << std::setw(6) << sortedIndex << " |";
 
-      st = OSS() << std::setw(7) << varianceOrder[i].first;
-      oss << st << " | ";
+      intermediateString = OSS() << " " << multiIndices;
+      oss << padString(intermediateString) << "|";
 
-      st = OSS() << multiIndices;
-      oss << st << String(st.size() < 29 ? 29 - st.size() : 0, ' ') << " | ";
-
-      st = OSS() << varianceOrder[i].second;
-      oss << st << "\n";
-    }
-    oss << String(60, '-') << "\n";
-    oss << "\n\n";
+      intermediateString = OSS() << " " << partOfVariance[sortedIndex];
+      oss << padString(intermediateString) << "|\n";
+    } // loop over the multi-indices
+    oss << "\n";
 
     // table of first/total order indices for each input
-    oss << String(60, '-') << "\n";
-    oss << "Component | Sobol' index           | Sobol' total index     \n";
-    oss << String(60, '-') << "\n";
+    // print table header
+    oss << "| Input |" << padString(" Name  ") << "|";
```

Comment:
Adding two blank spaces to the right is unnecessary, because the `padString` function will add more if necessary.
```suggestion
    oss << "| Input |" << padString(" Name ") << "|";
```